atlp-rwanda/ecommerce-app-champions-bn
# A RESTApi to get vendor Sales and buyer Orders

#### What does this PR do?

- Create RESTapi to get all seller sales
- Create RESTapi to get buyer orders
- The endpoint documented using swagger

#### Description of Task to be completed

- a vendor gets all the sales of the products sold
- a buyer gets all the orders of the products bought
- only the owner is allowed to update a product
#### How should this be manually tested?

- clone the repo
- switch to the fix-buyer-payment branch
- type npm install to install all packages
- npm run migrate:up and npm run db:seeds
- add STRIPE_SECRET_KEY = "sk_test_51MVA8uGaYgK6bC8Q7PwPCpkkktpHpvbt86v4cshRnJaSpWG8vjUhzND4Oxyd1rTHS0OV2bTNs9U72MnJGeEAzmBg00hZptGSrA" in .env file
- run the server by typing npm run dev
- in your browser navigate to swagger doc by typing localhost: PORT/docs/
- first add a product to the cart
- you either choose to update the cart by changing the quantity of the product
- after you can check out by navigating to the checkout endpoint
- copy the URL provided and past it in the browser to process the payment
- when the payment is successful, a receipt is send to the buyer via email.
- navigate to the sales endpoint on Swagger to get all vendor sales 
-  navigate to orders endpoint on Swagger to get all buyer orders
